Source/automap.cpp

```diff
@@ -344,14 +376,114 @@ void DrawRiverForkOut(const Surface &out, Point center, uint8_t color)
 	out.SetPixel({ center.x + AmLine(8), center.y + AmLine(16) - AmLine(4) }, color);
 }
 
-void DrawStairs(const Surface &out, Point center, uint8_t color)
+enum class StairsType : uint8_t {
+	Invalid,
+	DownRight,
+	DownLeft,
+	UpRight,
+	UpLeft,
+};
+
+StairsType GetStairsType(uint8_t tileId, uint8_t dlvl)
+{
+	if (dlvl == 0)
+		return StairsType::DownRight;
+	// clang-format off
+	switch (tileId) {
+	// Church
+	case 57:
+		return StairsType::DownRight;
+	case 66:
+		return StairsType::UpRight;
+	// Catacombs
+	case 78:
+		return StairsType::DownRight;
+	case 77:
+	case 160:
+		return StairsType::UpRight; // Shortcut
+	// Caves
+	case 47:
+		return StairsType::DownLeft;
+	case 48:
+		return StairsType::DownRight;

```

Comment:
The Crypt also uses 48, it's the only tile ID that overlaps across map types.

Reply:
I removed it as O changed which tile was considered the "primary" tile (stairs down)and subsequently removed everything that was a stairs part